build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -91,17 +93,23 @@ else()
   # Attempt to deduce a CUDAToolkit_ROOT (possibly downloading) if not
   # explicitly set.
   if(NOT CUDAToolkit_ROOT)
-    # Download the CUDA toolkit if needed.
-    fetch_cuda_toolkit()
-    if(CUDAToolkit_ROOT)
-      # Download succeeded.
-      message(STATUS "Using downloaded CUDA toolkit: ${CUDAToolkit_ROOT}")
-      set(CUDAToolkit_ROOT "${CUDAToolkit_ROOT}" PARENT_SCOPE)
-      # For some reason having the BIN_DIR set wrong can cause mayhem. Just make
-      # sure it is right.
-      set(CUDAToolkit_BIN_DIR "${CUDAToolkit_ROOT}/bin" PARENT_SCOPE)
+    if(CUDAToolkit_FOUND)
```

Comment:
It seems a bit weird that we have this same if-else condition in the main cmake file and here. Can we look for/download cuda all in the same place?

Reply:
We _could_, but this is currently set up to first check if CUDA is available and set the option defaults, then only actually download CUDA if the options are set.

```python
# ----------------------------------------------------------
# CMakeLists.txt

check_for_cuda()

if cuda available:
  default = on
else:
  default = off

option(cuda_compiler default)
option(cuda_runtime default)

if cuda_compiler or cuda_runtime:
  setup_cuda()

# ----------------------------------------------------------
# build_tools/third_party/cuda/CMakeLists.txt

setup_cuda():
  download_if_needed()
  # define a few variables
  # add a few libraries
```